WordPress/wporg-mu-plugins
# Add variations for header & footer styles

Fixes #221 — Added block variations and a `style` attribute to support the different color combinations. The default block shouldn't change, but now there are options for a "Black on white" and "White on blue." When selected, the style changes, updating the classes on the rendered group block, which controls how the custom properties are defined. I've also added filters so we can control this in PHP on classic themes if needed, `wporg_header_color_scheme` & `wporg_footer_color_scheme`.

<img width="352" alt="Screen Shot 2022-07-25 at 1 48 50 PM" src="https://user-images.githubusercontent.com/541093/180841761-37b4845d-d9e3-46d4-9601-1e2dbad842f4.png">

This also introduces a local layer of custom properties for the background, link color, and some related colors.

Since variations are registered in JS, I created two new folders `src/header` and `src/footer`, each with a `block.json`, and updated the PHP registration code to read from the `block.json`s.

While I was in the code, I also fixed #209, so the border is on the container `li` to span the width of the link text and dropdown button.

**Screenshots**

| Default | White on Blue | Black on White |
|---------|------------------|-----------------|
| ![default-default](https://user-images.githubusercontent.com/541093/180840200-03d56a62-5b39-4bac-aa7f-8ead36bdf154.png) | ![default-blue](https://user-images.githubusercontent.com/541093/180840198-c85dc300-18fb-44c7-8a59-6bd41e5ea59a.png) | ![default-white](https://user-images.githubusercontent.com/541093/180840204-72644401-dc55-4800-8878-9ad297511b1f.png) |

Active states— I combined a few screenshots in one. The Support submenu is hovered open. The Showcase link is focused. The search box is open. The Get WordPress button is hovered. **These (especially the white one) will need design input, but can be done in a followup PR.**

| Default | White on Blue | Black on White |
|---------|------------------|-----------------|
| ![active-states-default](https://user-images.githubusercontent.com/541093/180840193-f9058023-09d0-4fbd-bbd4-fc1bf7914a9e.png) | ![active-states-blue](https://user-images.githubusercontent.com/541093/180840190-c61dd063-716d-49d0-b6a9-a5a0257bf4aa.png) | ![active-states-white](https://user-images.githubusercontent.com/541093/180840194-33da71b9-7dfe-4b6e-8fe6-375b5b167e87.png) |

Small screen:

| Default | White on Blue | Black on White |
|---------|------------------|-----------------|
| ![menu-mobile-default](https://user-images.githubusercontent.com/541093/180840207-9695db6b-f950-4508-8665-cd69460a82e0.png) | ![menu-mobile-blue](https://user-images.githubusercontent.com/541093/180840205-d62b1b07-849f-4fb5-aae8-f1e2214cc623.png) | ![menu-mobile-white](https://user-images.githubusercontent.com/541093/180840208-cc86f2fc-1af8-483d-92ea-d975a0bfd5f9.png) |
| ![search-mobile-default](https://user-images.githubusercontent.com/541093/180840213-1702687f-cabd-418b-8260-6532148ffe3f.png) | ![search-mobile-blue](https://user-images.githubusercontent.com/541093/180840211-0e421022-0a59-4b1c-a712-eb25dd6b5308.png) | ![search-mobile-white](https://user-images.githubusercontent.com/541093/180840215-d9e20d3d-6dd9-404e-80ba-5d4d78327868.png) |

**To Test**

1. If you can, open the block in the site editor (maybe use the parent theme, since News doesn't work in the site editor)
2. Note, if you're trying to select the header, it's got 0 height because of the positioning used, so select it via the List view
3. Select a different variation
4. It should live-update in the editor
5. Save your changes & view on the frontend
6. The header and/or footer should match the style you've picked

Or, update the news template to manually add the attribute:

```html



```

Or, use the filter:

```php
add_filter( 'wporg_header_color_scheme', __NAMESPACE__ . '\update_color_scheme' );
add_filter( 'wporg_footer_color_scheme', __NAMESPACE__ . '\update_color_scheme' );

function update_color_scheme( $style ) {
	if ( ! $style ) {
		return 'white-on-blue';
	}
	return $style;
}
```
